pkg/apiserver/admission/admissiontimeout/timeoutadmission.go

```diff
@@ -34,18 +37,21 @@ func (p pluginHandlerWithTimeout) Admit(ctx context.Context, a admission.Attribu
 	go func() {
 		defer utilruntime.HandleCrash()
 		defer close(admissionDone)
-		admissionErr = mutatingHandler.Admit(ctx, a, o)
+		admissionErr = mutatingHandler.Admit(admissionCtx, a, o)
 	}()
 
 	select {
 	case <-admissionDone:
 		return admissionErr
-	case <-time.After(p.timeout):
-		return errors.NewInternalError(fmt.Errorf("admission plugin %q failed to complete mutation in %v", p.name, p.timeout))
+	case <-admissionCtx.Done():
+		return errors.NewInternalError(fmt.Errorf("admission plugin %q failed to complete mutation in %v", p.name, p.admissionTimeout))
```

Comment:
so i just realized that contexts are hierarchical - so if the parent context has a timeout of 5s and the child has a timeout of 13s, the child will timeout as soon as the parent times out after 5s.  so including `p.admissionTimeout` in the message may be misleading now.

we can  either remove the `admissionTimeout` from the error message or print the actual deadline of the context. thoughts?

Reply:
Indeed you are right https://github.com/golang/go/blob/6fefb7f9f3b632bdd0c3997ecc5b1096a5077cdf/src/context/context.go#L440. I learnt something new today.

Reply:
That information also concludes that nobody will not be able to increase admission timeout and will be able to decrease it.

Reply:
I think, it would be beneficial showing the timeout in error message. But it needs subtraction and some calculation because parent context has `time.Time` instead duration

Reply:
the problem I've found is that this will race on the error message if the handler implements the context too, because it should err with a "context cancelled" error. However, we need to keep this out of the goroutine for handlers that doesn't implement context cancellation, what I did is to wait a few milliseconds more for the later

https://github.com/openshift/library-go/pull/1244/files#diff-bfc004e22663aa56dcd945b47c1776b0beebfdf2eb176044cf6130faebc63a4c

Reply:
I don't have any inconvenience :)

Reply:
Thanks :)